lib/srv/app/aws/handler.go

```diff
@@ -55,23 +59,48 @@ type SignerHandlerConfig struct {
 	RoundTripper http.RoundTripper
 	// SigningService is used to sign requests before forwarding them.
 	*awsutils.SigningService
+	// CredentialsGetter is used to obtain STS credentials.
+	CredentialsGetter awsutils.CredentialsGetter
+	// Session is AWS session.
+	Session *awssession.Session
 }
 
 // CheckAndSetDefaults validates the AwsSignerHandlerConfig.
 func (cfg *SignerHandlerConfig) CheckAndSetDefaults() error {
-	if cfg.SigningService == nil {
-		return trace.BadParameter("missing SigningService")
-	}
+	var err error
 	if cfg.RoundTripper == nil {
-		tr, err := defaults.Transport()
+		cfg.RoundTripper, err = defaults.Transport()
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		cfg.RoundTripper = tr
 	}
 	if cfg.Log == nil {
 		cfg.Log = logrus.WithField(trace.Component, "aws:signer")
 	}
+	if cfg.CredentialsGetter == nil {
+		cfg.CredentialsGetter, err = awsutils.NewCachedCredentialsGetter(awsutils.CachedCredentialsGetterConfig{})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	if cfg.Session == nil {
+		cfg.Session, err = awssession.NewSessionWithOptions(awssession.Options{
+			SharedConfigState: awssession.SharedConfigEnable,
+		})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	if cfg.SigningService == nil {
```

Comment:
Can we hide `SigningService` and always initialize it here? This would ensure consistency in how it is initialized.